File: Backend/Core/src/main/scala/Objects/CurriculumService.scala
package Objects

case class Curriculum(
    stage: Stage,
    recommendations: List[Recommendation]
)

case class Recommendation(
    id: String,
    subject: Subject,
    priority: Int
)

case class Subject(
    id: String,
    name: String,
    credits: Int
)

case class Teaching(
    subject: Subject,
    lectures: List[Lecture]
)

case class Lecture(
    id: String,
    term: Term,
    location: String,
    schedule: String,
    capacity: Int
)

case class Course(
    id: String,
    teacher: User,
    location: String,
    schedule: String,
    capacity: Int
)

case class Comment(
    id: String,
    content: String
)

case class Review(
    term: Term,
    comments: List[Comment]
)

File: Frontend/types/curriculumService.ts
import { Stage, Term } from "./registerService";
import { User } from "./userService";

export interface Curriculum {
  stage: Stage;
  recommendations: Recommendation[];
}

export interface Recommendation {
  id: string;
  subject: Subject;
  priority: number;
}

export interface Subject {
  id: string;
  name: string;
  credits: number;
}

export interface Teaching {
  subject: Subject;
  lectures: Lecture[];
}

export interface Lecture {
  id: string;
  term: Term;
  location: string;
  schedule: string;
  capacity: number;
}

export interface Course {
  id: string;
  teacher: User;
  location: string;
  schedule: string;
  capacity: number;
}

export interface Comment {
  id: string;
  content: string;
}

export interface Review {
  term: Term;
  comments: Comment[];
}
